pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -137,9 +138,32 @@ def swa_end(self) -> int:
     def pl_module_contains_batch_norm(pl_module: 'pl.LightningModule'):
         return any(isinstance(module, nn.modules.batchnorm._BatchNorm) for module in pl_module.modules())
 
+    @contextmanager
+    def _prevent_data_deepcopy(self, pl_module: 'pl.LightningModule'):
+        """
+        This function is used to prevent deepcopy of dataloaders / trainer while deepcopying the ``LightningModule``.
+        """
+        train_dataloader = pl_module.train_dataloader
+        val_dataloader = pl_module.val_dataloader
+        test_dataloader = pl_module.test_dataloader
+        predict_dataloader = pl_module.predict_dataloader
+        trainer = pl_module.trainer
+        pl_module.train_dataloader = None
+        pl_module.val_dataloader = None
+        pl_module.test_dataloader = None
+        pl_module.predict_dataloader = None
+        pl_module.trainer = None
+        yield
+        pl_module.train_dataloader = train_dataloader
+        pl_module.val_dataloader = val_dataloader
+        pl_module.test_dataloader = test_dataloader
+        pl_module.predict_dataloader = predict_dataloader
+        pl_module.trainer = trainer
```

Comment:
We could do this by defining `__getstate__` on the `LightningModule` instead

Reply:
Not sure we want to, some users might expect a complete deepcopy. At least, it is contained to the SWA Callback for now.

Reply:
@justusschock @awaelchli  Any opinions ?

Reply:
what else could be problematic for deepcopy in LightningModule? User might have some problematic objects?
Would they be able to get around it by implementing getstate?

Reply:
I don't have any strong opinions who this should be handled

Reply:
I think this would also be beneficial for the `ModelPruning` callback which also deepcopies the model. This is why I pointed it out.

Reply:
Done !